File: BackJoon/5532/main.py
vacation= int(input())
math_page= int(input())
korean_page=int(input())
math_study=int(input())
korean_study=int(input())

math_day= math_page//math_study
korean_day= korean_page//korean_study

if math_page%math_study!=0:
    math_day += 1

if korean_page%korean_study!=0:
    korean_day += 1

print(vacation-max(math_day,korean_day))
    

File: BackJoon/5532/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);

        int vacation = sc.nextInt();
        int korean = sc.nextInt();
        int math = sc.nextInt();
        int Sang_korean = sc.nextInt();
        int Sang_math = sc.nextInt();

        int korean_study =  korean/Sang_korean;
        int math_study = math/Sang_math;

        if(korean % Sang_korean!=0){
            korean_study += 1;
        }
        if (math % Sang_math!=0){
            math_study += 1;
        }
    System.out.println(vacation-Math.max(korean_study,math_study));
    }
}
